packages/yarnpkg-fslib/sources/algorithms/copyPromise.ts

```diff
@@ -25,7 +25,13 @@ export async function copyPromise<P1 extends Path, P2 extends Path>(destinationF
   const prelayout: Operations = [];
   const postlayout: Operations = [];
 
-  await destinationFs.mkdirPromise(destination, {recursive: true});
+  try {
+    await destinationFs.mkdirPromise(destination, {recursive: true});
```

Comment:
Isn't the better fix to just call `mkdirPromise` on the dirname of `destination`?

Reply:
Oh, right. I missed the fact that we call `destinationFs.mkdirPromise(destination, {mode: sourceStat.mode})` further down. Fixed!